core/utils/pcap_handle_test.cc

```diff
@@ -0,0 +1,67 @@
+// Some basic unit tests for PcapHandle class -- complete testing requires
+// integration testing to make sure it can actually talk to driver.
+
+// TODO: Integration tests later for Recv, Send, Reset, and dev constructor.
+
+#include "pcap_handle.h"
+#include <gtest/gtest.h>
+#include <string>
+
+class PcapHandleFixtureTest : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    // Can't actually set up port w/o integration test so fake internal state
+    pcap_with_fake_handle.handle_ = (pcap_t*)342;
+  }
+
+  virtual void TearDown() {
+    // So destructor doesn't try to close fake connection.
+    pcap_with_fake_handle.handle_ = nullptr;
+  }
+
+  PcapHandle pcap_with_fake_handle;
+};
+
+// With no parameters, should always be uninitialized.
+TEST(PcapHandleBasicTest, EmptyConstructor) {
+  PcapHandle p;
+  ASSERT_FALSE(p.is_initialized());
+  ASSERT_NE(0, p.SendPacket(nullptr, 44));
+  int caplen = 72;
+  ASSERT_EQ(nullptr, p.RecvPacket(&caplen));
+  ASSERT_EQ(0, caplen);
+}
+
+// Should result in an uninitialized device if binding to device fails.
+TEST(PcapHandleBasicTest, BadDevice) {
+  PcapHandle p("sangjinhan");
+  ASSERT_FALSE(p.is_initialized());
+  ASSERT_NE(0, p.SendPacket(nullptr, 44));
+  int caplen = 72;
+  ASSERT_EQ(nullptr, p.RecvPacket(&caplen));
+  ASSERT_EQ(0, caplen);
+}
+
+// Check that it knows it's initialized if it has a pcap handle.
+TEST_F(PcapHandleFixtureTest, IsInitialized) {
+  ASSERT_TRUE(pcap_with_fake_handle.is_initialized());
+}
+
+// Move operator.
+TEST_F(PcapHandleFixtureTest, MoveOperator) {
+  ASSERT_TRUE(pcap_with_fake_handle.is_initialized());
+  PcapHandle MoveTo = std::move(pcap_with_fake_handle);
```

Comment:
Unless I'm reading this wrong, this will actually invoke the move constructor, not the move assignment operator because the object is being created on this line.  However if you were to separate the lines as:

PcapHandle MoveTo();
MoveTo = std::move(pcap_with_fake_handle);

then it will invoke the move assignment operator.

The way I think about it is this:
Does the object on the left already exist? Yes: it will use an assignment operator, No: a constructor.
Does the object on the right already exist? Yes: it will not move unless told to explicitly, No: it will try to use a move if possible